src/drivers/stm32/drv_io_timer.c

```diff
@@ -376,14 +380,39 @@ static int timer_set_rate(unsigned timer, unsigned rate)
 {
 
 	/* configure the timer to update at the desired rate */
-	rARR(timer) = 1000000 / rate;
+	rARR(timer) = timer_freq[timer] * 1000000 / rate;
```

Comment:
@kd0aij the rARR and most registers are 16 bit[1]. 

When  timer_freq[timer]  = 8 this will overflow.

[1] See AN4013 Application note - DocID022500 Rev 3 for what CPU and what timers have 16/32 bit registers.